docs/src/index.md

```diff
@@ -0,0 +1,457 @@
+`Symbol`s and strings. The following will fail:
+
+```julia
+df = DataFrame(A = 1:3, B = [2, 1, 2])
+@eachrow df begin 
+    :A = cols(2)
+end
+
+@with df begin 
+    cols(1) + cols("A")
+end
+```
+
+while the following will work without error
+
+```julia
+@eachrow df begin 
+    cols(1) = cols(2)
+end
+
+@with df begin 
+    cols(1) + cols(2)
+end
+```
+
+Note that `cols` is *not* a standard Julia function. It is only used to modify the 
+way that macros in DataFramesMeta.jl escape arguments and has no behavior of its own 
+outside of DataFramesMeta macros.
+
+
+# Working with `Symbol`s without referring to columns
+
+To refer to `Symbol`s without aliasing the column in a data frame, use `^`. 
+
+```
+df = DataFrame(x = [1, 1, 2, 2], y = [1, 2, 101, 102]);
+@select(df, :x2 = :x, :x3 = ^(:x))
+```
+
+This rule applies to all DataFramesMeta macros.
+
+## Comparison with `dplyr` and LINQ
+
+A number of functions for operations on DataFrames have been defined.
+Here is a table of equivalents for Hadley's
+[dplyr](https://github.com/hadley/dplyr) and common
+[LINQ](http://en.wikipedia.org/wiki/Language_Integrated_Query)
+functions.
+
+    Julia             dplyr            LINQ
+    ---------------------------------------------
+    @where            filter           Where
+    @transform        mutate           Select (?)
+    @by                                GroupBy
+    groupby           group_by         GroupBy
+    @combine          summarise/do
+    @orderby          arrange          OrderBy
+    @select           select           Select
+
+
+## `@linq` and other chaining macros
+
+There is also a `@linq` macro that supports chaining and all of the
+functionality defined in other macros. Here is an example of `@linq`:
+
+```julia
+df = DataFrame(a = repeat(1:5, outer = 20),
+               b = repeat(["a", "b", "c", "d"], inner = 25),
+               x = repeat(1:20, inner = 5))
+
+x_thread = @linq df |>
+    transform(y = 10 * :x) |>
+    where(:a .> 2) |>
+    by(:b, meanX = mean(:x), meanY = mean(:y)) |>
+    orderby(:meanX) |>
+    select(:meanX, :meanY, var = :b)
+```
+
+Relative to the use of individual macros, chaining looks cleaner and
+more obvious with less noise from `@` symbols. This approach also
+avoids filling up the limited macro name space. The main downside is
+that more magic happens under the hood.
+
+This method is extensible. Here is a comparison of the macro and
+`@linq` versions of `with`.
+
+```julia
+function linq(::DataFramesMeta.SymbolParameter{:with}, d, body)
+    with_helper(d, body)
+end
+
+df = DataFrame(A = 1:3, B = [2, 1, 2])
+
+@linq df |>
+    with(:A .* :B)
+```
+
+The `linq` method above registers the expression-replacement method
+defined for all `with()` calls. It should return an expression like a
+macro.
+
+Again, this is experimental. Based on feedback, we may decide to only
+use `@linq` or only support the set of linq-like macros.
+
+Alternatively you can use Lazy.jl `@>` macro like this:
```

Comment:
do we really need to show Lazy.jl here? What is the value of this?

Reply:
I agree. I'm fine to delete.